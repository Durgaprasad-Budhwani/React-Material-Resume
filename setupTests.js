/**
 * Created by Durgaprasad Budhwani on 12/14/2017.
 */
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Element.prototype.getBoundingClientRect = function(){
    return {
        y: -100,
        top: -100
    }
};

configure({ adapter: new Adapter() });