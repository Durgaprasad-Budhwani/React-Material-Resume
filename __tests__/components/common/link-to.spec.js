/**
 * Created by Durgaprasad Budhwani on 12/13/2017.
 */
/* global expect,it, describe */
import React from 'react';
import LinkToContainer, {LinkTo} from '../../../src/components/common/link-to';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import toJSON from 'enzyme-to-json';
import '../../../setupTests';

describe('<LinkTo />', () => {
    const classes = {
        scrollChor: {},
        navigationItem: {},
        spanClass: {},
        activeSpanClass: {},
        activeTitle: {}
    };
    
    it('<LinkToContainer />', () => {
        const tree = renderer.create(<LinkToContainer
            to="to"
            title="title"
            enableNavigationItemClass={true}
            offset={100}
            classes={{}}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('<LinkTo> handlerSetInactive', () => {
        const wrapper = mount(<LinkTo
            to="to"
            title="title"
            enableNavigationItemClass={true}
            offset={100}
            classes={classes}
        />);
        const instance = wrapper.instance();
        instance._handleSetInactive();
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    
    it('<LinkTo> handlerSetInactive', () => {
        const wrapper = mount(<LinkTo
            to="to"
            title="title"
            enableNavigationItemClass={true}
            offset={100}
            classes={classes}
        />);
        const instance = wrapper.instance();
        instance._handleSetActive();
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});


