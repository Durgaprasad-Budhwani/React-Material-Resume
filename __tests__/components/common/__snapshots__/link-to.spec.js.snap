// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkTo /> <LinkTo> handlerSetInactive 1`] = `
<LinkTo
  classes={
    Object {
      "activeSpanClass": Object {},
      "activeTitle": Object {},
      "navigationItem": Object {},
      "scrollChor": Object {},
      "spanClass": Object {},
    }
  }
  enableNavigationItemClass={true}
  offset={100}
  title="title"
  to="to"
>
  <Link
    className="[object Object]"
    duration={500}
    offset={100}
    onSetActive={[Function]}
    onSetInactive={[Function]}
    smooth={true}
    spy={true}
    to="to"
  >
    <div>
      <span
        className=""
      >
        title
      </span>
      <span
        className=""
      />
    </div>
  </Link>
</LinkTo>
`;

exports[`<LinkTo /> <LinkTo> handlerSetInactive 2`] = `
<LinkTo
  classes={
    Object {
      "activeSpanClass": Object {},
      "activeTitle": Object {},
      "navigationItem": Object {},
      "scrollChor": Object {},
      "spanClass": Object {},
    }
  }
  enableNavigationItemClass={true}
  offset={100}
  title="title"
  to="to"
>
  <Link
    className="[object Object]"
    duration={500}
    offset={100}
    onSetActive={[Function]}
    onSetInactive={[Function]}
    smooth={true}
    spy={true}
    to="to"
  >
    <div>
      <span
        className=""
      >
        title
      </span>
      <span
        className=""
      />
    </div>
  </Link>
</LinkTo>
`;

exports[`<LinkTo /> <LinkToContainer /> 1`] = `
<div>
  <span
    className=""
  >
    title
  </span>
  <span
    className="LinkTo-spanClass-3"
  />
</div>
`;
